bolt-taproot-gossip.md

```diff
@@ -0,0 +1,748 @@
+
+The sender:
+
+  - if `option_schnorr_gossip` was negotiated: 
+    - MAY set the `announce_channel` bit in `channel_flags`
+  - otherwise:
+    - MUST NOT set the `announce_channel` bit.
+
+The receiving node MUST fail the channel if:
+
+  - if `option_schnorr_gossip` was not negotiated and the `announce_channel` 
+    bit in the `channel_flags` was set.
+
+### `channel_ready` Extensions
+
+These extensions only apply if the `option_taproot` channel type was set in the
+`open_channel` message along with the `announce_channel` channel flag.
+
+1. `tlv_stream`: `channel_ready_tlvs`
+2. types:
+   1. type: 0 (`announcement_node_pubnonce`)
+   2. data:
+      * [`66*byte`: `public_nonce`]
+   3. type: 1 (`announcement_bitcoin_pubnonce`)
+   4. data:
+      * [`66*byte`: `public_nonce`]
+   
+### Requirements
+
+The sender: 
+
+  - MUST set the `announcement_node_pubnonce` and 
+    `announcement_bitcoin_pubnonce` tlv fields in `channel_ready`.
+  - SHOULD use the MuSig2 `NonceGen` algorithm to generate two unique secret 
+    nonces and use these to determine the corresponding public nonces: 
+    `announcement_node_pubnonce` and `announcement_bitcoin_pubnonce`.
+  
+The recipient: 
+
+  - MUST fail the channel if: 
+    - if the `announcement_node_pubnonce` or `announcement_bitcoin_pubnonce` tlv 
+      fields are missing.
+    - if the `announcement_node_pubnonce` and `announcement_bitcoin_pubnonce` 
+      are equal.
+    
+### Rationale
+
+The final signature included in the `channel_announcement_v2` message will be a
+single Schnorr signature that needs to be valid for a public key derived by 
+aggregating all the `node_ID` and `bitcoin_key` public keys. The signature for 
+this key will thus be created by aggregating (via MuSig2) four partial 
+signatures. One for each of the keys: `node_ID_1`, `node_ID_2`, `bitcoin_key_1`
+and `bitcoin_key_2`. A nonce will need to be generated and exchanged for each
+of the keys and so each node will need to send the other node two nonces. The
+`announcement_node_nonce` is for `node_ID_x` and the`announcement_bitcoin_nonce`
+is for `bitcoin_key_x`.
+
+Since the channel can only be announced once the `channel_ready` messages have 
+been exchanged and since it is generally preferred to keep nonce exchanges as 
+Just In Time as possible, the nonces are exchanged via the `channel_ready` 
+message.
+
+## The `announcement_signatures_2` Message
+
+Like the legacy `announcement_signatures` message, this is a direct message
+between the two endpoints of a channel and serves as an opt-in mechanism to
+allow the announcement of the channel to the rest of the network. 
+
+1. type: xxx (`announcement_signatures_2`)
+2. data (tlv_stream):
+   1. type: 0 (`channel_id`)
+   2. data:
+       * [`channel_id`:`channel_id`]
+   3. type: 1 (`short_channel_id`)
+   4. data:
+       * [`short_channel_id`:`short_channel_id`]
+   5. type: 2 (`partial_signature`)
+   6. data:
+       * [`partial_signature`:`partial_signature`]
```

Comment:
I would definitely not use a TLV for channel_id.  Every other message starts with channel_id, we should keep it fixed.  The other two fields, well, they're OK, but it seems weird that they're optional.